.github/workflows/jgiven_branch_and_pr_build.yml

```diff
@@ -0,0 +1,27 @@
+name: Java CI with Gradle
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build JGiven on Java ${{ matrix.Java }} with Junit ${{matrix.junit}}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ '8', '11', '13']
+        junit: [ '4.9', '4.13' ]
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK ${{matrix.java}}
+        uses: actions/setup-java@v2
+        with:
+          java-version: ${{matrix.java}}
+          distribution: 'adopt'
+      - name: Grant execute permission for gradlew
+        run: chmod +x gradlew
+      - name: Build with Gradle  and Junit ${{matrix.junit}}
+        env:
+          JUNIT_VERSION: ${{matrix.junit}}
+        run:   "./gradlew -i -s check -DjunitVersion=${JUNIT_VERSION}"
+      - name: Run Jacoco
+        if: ${{matrix.java == '8' && github.ref == 'refs/heads/master' && github.head_ref == null}}
```

Comment:
This might be due to the general syntax of .yml files, but somehow this condition seems odd - shouldn't it be something like 

`if: (${{matrix.java}} == '8' && blabla )` ?

Reply:
https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions recommends using this syntax to have this evaluated as expression

Reply:
This is something one needs to get used to, but in this case it's fine.